src/language/records.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Records
+description: Summary of the record data structure in Dart.
+toc: false
+---
+
+Records are an anonymous, immutable, aggregate type. Like other [collection types][], 
+they let you bundle multiple objects into a single object. Unlike other collection 
+types, records are fixed-sized, heterogeneous, and typed.
+
+Records are real values; you can store them in variables, pass them to and from
+functions, and store them in lists.
+
+## Record syntax
+
+Records are comma-delimited lists of named or positional fields,
+enclosed in parenthesis:
+
+```dart
+var record = ('first', a: 2, b: true, 'last');    // Record expression
+```
+
+Named fields inside record type annotations go inside a brace-delimited section
+of type and name pairs:
+
+```dart
+(Sring, {int a, bool b}, String?) record;    // Record type annotation
+```
```

Comment:
The comments in the code samples help, but I worry this is still too subtle of a jump from a record *expression* to a record *type annotation*. Because it jumps from positional fields to named fields, it's easy to overlook that it's also moving from an expression to a type annotation too.

I'd probably insert an example of a record type annotation with only positional fields first and explain that it's a type annotation and where you might use it.

Reply:
For what it's worth, I always use "type annotation" to mean the syntax for how to write down a type. They aren't optional "annotations" in the Dart 1 sense, but as far as I know, "type annotation" is pretty standard across languages as a way to refer to type terms.

Reply:
> I'd probably insert an example of a record type annotation with only positional fields first and explain that it's a type annotation _and where you might use it._

@munificent Where _would_ you use one? 
Do they only come up for destructuring? 

I'm drawing a blank [from the spec](https://github.com/dart-lang/language/blob/master/accepted/future-releases/records/records-feature-specification.md#record-type-annotations). I was going to say, "to declare a record", but that doesn't sound right, since they're immutable it's not like you can declare a record then populate it later (can you?)

[Even for typing](https://github.com/dart-lang/language/blob/master/accepted/future-releases/records/records-feature-specification.md#record-type-annotations:~:text=(num%2C%20num)%20r%20%3D%20(3%2C%203.5)..%241.isEven%3B), it seems like the types are going to be inferred regardless of what you write in a type annotation, so that doesn't seem like a use case, just an example for the sake of the spec.

Reply:
Yes, you rarely need them for variable declarations because they can be inferred. But they come up at API boundaries: Parameter types and return types. For example, here's a function that takes a record and swaps its fields:

```dart
(int, int) swap((int, int) record) {
  var (a, b) = record;
  return (b, a);
}
```

The two `(int, int)` here are record type annotations.